ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -675,6 +684,33 @@ public void shutdown() throws StoreException {
     }
   }
 
+  /**
+   * On an exception/error, if error count exceeds threshold, properly shutdown store.
+   */
+  private void onError() throws StoreException {
+    int count = errorCount.incrementAndGet();
+    if (count == config.storeIOErrorCountToTriggerShutdown) {
+      logger.info("Shutting down BlobStore {} because IO error count exceeds threshold", storeId);
+      shutdown();
+    }
+  }
+
+  /**
+   * If store restarted successfully or at least one operation succeeded, reset the error count.
+   */
+  private void onSuccess() {
+    if (errorCount.getAndSet(0) >= config.storeIOErrorCountToTriggerShutdown) {
```

Comment:
I don't understand the `if` here. If the errorCount() was higher or equal to `config.storeIOErrorCountToTriggerShutdown`,  won't the store be shutting/shut down?